org.eclipse.buildship.ui/src/main/java/org/eclipse/buildship/ui/view/task/GroupProjectsAction.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2018 the original author or authors.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+
+package org.eclipse.buildship.ui.view.task;
+
+import com.google.common.base.Preconditions;
+
+import org.eclipse.buildship.ui.PluginImage;
+import org.eclipse.buildship.ui.PluginImages;
+import org.eclipse.jface.action.Action;
+
+/**
+ * An action on the {@link TaskView} that toggles project grouping.
+ *
+ * @author Charles Wu <charles.wu@liferay.com> 
+ */
+public final class GroupProjectsAction extends Action {
```

Comment:
Rename to `ShowFlattenedProjectHierarchy` and verify that the implementation matches the naming.